### PR TITLE
Fixed incorrect normalization

### PR DESCRIPTION
**Pull request by audioXD**
I've notices a possible typo in [s`rc/transformers/image_processing_utils_fast.py#compile_friendly_resize`](https://github.com/huggingface/transformers/blob/c81723d31b8a532803feee2bbce66e4f03829759/src/transformers/image_processing_utils_fast.py#L307) for `uin8` the normalization is done slightly off with `256` instead of `255`, which still works because its done consistenly (its normalized and denormalized the same way) incorrect.

[image = image.float() / 256](https://github.com/huggingface/transformers/blob/c81723d31b8a532803feee2bbce66e4f03829759/src/transformers/image_processing_utils_fast.py#L307)
[image = image * 256](https://github.com/huggingface/transformers/blob/c81723d31b8a532803feee2bbce66e4f03829759/src/transformers/image_processing_utils_fast.py#L309C13-L309C32)

The exaplanation is simple:

`255` (the max value for a uint8) should map to `1.` but it doesn't with the current implementation

### CODE REVIEW

**Comment by Rocketknight1** [comment]
cc @yonigozlan @qubvel

**Review by qubvel — APPROVED** [review]
Thanks, makes sense 👍

**Comment by yonigozlan** [comment]
@remi-or I think you had used 256 on purpose? Can you check that this changes isn't breaking?

**Comment by remi-or** [comment]
@yonigozlan I think I tested it against `torchvision.resize` and parity test worked with `256`. Maybe the snippet is in the PR? 
edit: PR is here https://github.com/huggingface/transformers/pull/38540

File: src/transformers/image_processing_utils_fast.py [hunk 1]
```diff
@@ -304,9 +304,9 @@ def compile_friendly_resize(
         A wrapper around `F.resize` so that it is compatible with torch.compile when the image is a uint8 tensor.
         """
         if image.dtype == torch.uint8:
-            image = image.float() / 256
+            image = image.float() / 255
             image = F.resize(image, new_size, interpolation=interpolation, antialias=antialias)
-            image = image * 256
+            image = image * 255
             image = torch.where(image > 255, 255, image)
             image = torch.where(image < 0, 0, image)
```
Comment (qubvel) [inline src/transformers/image_processing_utils_fast.py:311]: btw, it might be more optimal to use `torch.clamp(image, 0, 255)` once instead of `torch.where` twice

Comment (remi-or) [reply src/transformers/image_processing_utils_fast.py:311]: Normally I would agree! But this is on purpose : https://github.com/huggingface/transformers/pull/38540

Comment (qubvel) [reply src/transformers/image_processing_utils_fast.py:311]: Ok, according to the PR it seems we have to revert this PR to use 256 and keep torch.where. To prevent this code from further regression we have to either add tests that fails on CI (cuda) if modified or **properly** comment the code

Comment (remi-or) [reply src/transformers/image_processing_utils_fast.py:311]: The regression this PR introduced already caused failures on the [AMD CI](https://github.com/huggingface/transformers/actions/runs/17391751450/job/49367005103), which is as important as NVIDIA (or cuda) CI!
As for **properly** commenting the code, both code paths where `compile_friendly_resize` is called are commented. You can check it out by expanding the diff, those lines are right above the function :slightly_smiling_face: 
If you want, we can add `# this is to match torchvision.resize` next to 256 and `# We use torch.where instead of torch.clamp to avoid an error with torch.compile` as comments to make sure no one will introduce the regression again. Wdyt?

Comment (qubvel) [reply src/transformers/image_processing_utils_fast.py:311]: @remi-or absolutely agree AMD CI is as important as NVIDIA, what I'm trying to say is that we need a test that fails in PR's CI to prevent merging this PR. In terms of comments, yeah, it's better to comment non-obvious code right in place, otherwise it looks like a typo and is easy to miss the comment located in a different part (and that's happened in this case).
I'll do a quick fix for this, thanks for jumping in and clarifying 🤗
